packages/react-native/ReactCommon/react/runtime/TimerManager.cpp

```diff
@@ -258,49 +258,36 @@ void TimerManager::attachGlobals(jsi::Runtime& runtime) {
   }
 
   runtime.global().setProperty(
-      runtime,
-      "setTimeout",
-      jsi::Function::createFromHostFunction(
-          runtime,
-          jsi::PropNameID::forAscii(runtime, "setTimeout"),
-          3, // Function, delay, ...args
-          [this](
-              jsi::Runtime& rt,
-              const jsi::Value& thisVal,
-              const jsi::Value* args,
-              size_t count) {
+    runtime,
+    "setTimeout",
+    jsi::Function::createFromHostFunction(
+        runtime,
+        jsi::PropNameID::forAscii(runtime, "setTimeout"),
+        3, // Function, delay, ...args
+        [this](jsi::Runtime& rt, const jsi::Value& thisVal, const jsi::Value* args, size_t count) {
             if (count == 0) {
-              throw jsi::JSError(
-                  rt,
-                  "setTimeout must be called with at least one argument (the function to call).");
+                throw jsi::JSError(rt, "setTimeout must be called with at least one argument (the function to call).");
             }
 
-            if (!args[0].isObject() || !args[0].asObject(rt).isFunction(rt)) {
-              throw jsi::JSError(
-                  rt, "The first argument to setTimeout must be a function.");
-            }
-            auto callback = args[0].getObject(rt).getFunction(rt);
+            auto callback = args[0].isObject() && args[0].asObject(rt).isFunction(rt)
+                                ? args[0].getObject(rt).getFunction(rt)
+                                : jsi::Function::createFromHostFunction(
```

Comment:
Creating a host function for invalid calls seems wasteful and I don't think it's necessary. I saw that Chrome returns `0` for some types of calls to `setTimeout` (e.g.: `setTimeout('')`) so we could do that same here (just return `jsi::Value(0)`.

You need to make another change to make that work though. In `TimerManager.h` we initialize the value with `0` (as the initial timer handle) and we should change that to 1, so the first timer doesn't conflict with an "invalid" one. When you do that, might be worth renaming `timerIndex_` as `nextTimerHandle_` for clarity.

This is another aspect where we'll align with the spec: https://html.spec.whatwg.org/multipage/timers-and-user-prompts.html#timer-initialisation-steps (the handle must be a value greater than 0, but we can use 0 for the invalid ones).